sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/ECDsaFactory.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+using Azure.Core;
+
+namespace Azure.Security.KeyVault.Keys.Cryptography
+{
+    /// <summary>
+    /// Extensions for creating ECDsa from a Key Vault client.
+    /// </summary>
+    public static class ECDsaFactory
```

Comment:
Talking with @schaabs offline some (one of our architects and who worked on Key Vault originally), we think it would actually make more sense to hide the public classes you have now and add methods - names certainly open for discission - like `RSA CryptographyClient.ToRSA()`, `ECDsa CryptographyClient.ToECDsa()`, etc. Perhaps even some helps on `KeyClient` that would internally use the new `GetCryptographyClient()` I added recently since it shares the relatively expensive pipeline. But the implementation would remain private.

@KrzysztofCwalina @tg-msft what do you think about this design?

Reply:
I like the approach.

Why are we returning the base `RSA` class instead of our own `KeyVaultRSA` class though?  If we don't want to return `KeyVaultRSA` because it doesn't have any other members, let's just make the type `internal` then.

I also think we'd want slightly more verbose method names because `ToRSA` doesn't feel descriptive enough _(but I'm  someone who hated that we used crypto algorithm acronyms as class names in the first place...)_.  Maybe something as simple as `KeyVaultRSA CryptographyClient.CreateKeyVaultRSA(...)`?

Reply:
I see little value in returning our own type. Maybe we could hang a `Uri KeyId { get; }` off it, but probably not a lot of value in this since the purpose is to create a crypto provider primitive for .NET, so I doubt it'd ever get used. That said, at least it opens up the possibility of adding things. I wouldn't be against returning something like a `KeyVaultRSA` or `KeyVaultECDsa`.

I agree on the names. Like I said, they are certainly open for discussion. They seem to short, but I'm not sure what to add. Even .NET is inconsistent, where the `RSACryptoServiceProvider` equivalent for EC is just `ECDsa` (also equivalent to `RSA`, which is recommended for cross-plat development). But the few options I've considered I think are worse, like "(To|Get)(RSA|ECDsa)CryptoProvider". Given my suggest return types - the primitives @clairernovotny is exposing - the "To(RSA|ECDsa)" at least makes sense, given the return an `RSA` or `ECDsa`.

Reply:
One of the reasons the type is public is that it may be necessary to check for a specific provider in some contexts:

https://github.com/dotnet/SignService/blob/main/src/SignService/Utils/mansign.cs#L510-L522

If there's a better way, then I'm open.

Reply:
To what else could `signer.StrongNameKey` be set? That is, why can't it just be an `RSA`? `RSA` can sign and, if no private key, will throw - same as any implementation using `CryptographyClient` would do.

Reply:
To be honest, it's been a long time since I touched that code. I believe it was to isolate/minimize the behavioral changes in case of side effects.

Reply:
Another reason the type is public is so that they can be created with a shared `KeyVaultContext`. The idea was that the `KeyVaultContext` struct contained the public key and certificate. That data is usually required alongside the algorithms, such as when doing anything with SignedCms, XML DSig, as the cert is a core part of things.

Reply:
I've forgotten exactly which types being public we were talking about. This comment started on the static factory classes, but which type were we talking about that needed to be public for the last two comments you made, @clairernovotny?

Reply:
I believe we were talking about the implementation types -- `KeyVaultRSA` and `KeyVaultECDsa` -- do we return those or return `RSA` and `ECDsa` instead?